### DISCUSSION TITLE
How would I generate API keys for a user?

### DISCUSSION THREAD

**Discussion by FelixZY** [nextauthjs/next-auth#7263] [Help]
I'm developing an API to be used by third parties. The flow is:
1. User signs up with email+password/Google/etc.
2. User logs into a "developer portal" (jwt based)
3. User can create a long-lived API key. (How?)
4. User can call my API from their application using API key instead of email+password/Google/etc.

Some requirements for the API key:
* One user can have many api keys
* Each API key is associated with a single account
* API keys must be revokable
* API keys should not expire or require renewal
* I want to store additional information relating to the keys in my database (e.g. logging, permissions)

Is there a way to create such keys in a way compatible with next-auth/auth.js?

**Comment by vmiguellima** [comment]
You could take inspiration from https://github.com/calcom/cal.com/blob/main/packages/prisma/schema.prisma

**Reply by FelixZY** [reply]
Hm, yes; looks like a good reference project. Thank you!
